### ISSUE TITLE
Requesting less indent of what should be in list long assignment

### ISSUE THREAD

**Issue by latot** [r-lib/lintr#2524]
Hi, this is weird, ```lintr``` is requesting less indent that should be:

```R
library(wrapr)

list(
  "segment" =
    c(
      1,
      2
    ) %.>%
      sum(.)     
)
```

At least in my experience, and evaluations of other codes, I'm using the right way the indent, actually use the same format out of a list does not trigger any message, is only there.

Error:

```R
/tmp/c.R:7:6: style: [indentation_linter] Indentation should be 4 spaces but is 6 spaces.
      sum(.)
```

Note: ```wrapr``` is smilar to ```magrittr``` we just write explicitly where goes the previous value.

Thx!

**Comment by MichaelChirico** [comment]
the lint looks correct to me:

```r
foo(
) %>%
  bar()

foo() %>%
  bar()
```

%>% induces a new indentation for the next line in the tidyverse guide

**Comment by latot** [comment]
!!?

Actually, what I wrote above in your example is exactly that:

```
foo() %>%
  bar()
```

What ```lintr``` request to have is:

```
foo() %>%
bar()
```

**Comment by rtpop** [comment]
I'm having a similar issue with functions. For some reason, it seems not to recognise that some code is inside a function and will complain about the indenting. But what is even weirder is that in some files, it works fine while in others it doesn't. I've been unable to figure out what the difference is or why this happens. But it's very irritating.

**Comment by AshesITR** [comment]
@rtpop Thanks for reporting this issue. You need to use block code fences (```` ``` ````) for multi-line code instead of inline code markers (`` ` ``) if you want line breaks to appear.

Please file a separate issue with a reproducible example so that the problem can be tracked and discussed separately.

**Comment by rtpop** [comment]
@AshesITR Thanks!

**Comment by MichaelChirico** [comment]
For the original example, we can trace it to having `=` treated as assignment and `assignment_as_infix=TRUE` by default:

```
str <- '
list(
  "segment" =
    c(
      1,
      2
    ) %.>%
      sum(.)     
)
'

lint(text = str, linters = list(T = indentation_linter(assignment_as_infix=TRUE), F=indentation_linter(assignment_as_infix=FALSE)))
# <text>:8:6: style: [T] Indentation should be 4 spaces but is 6 spaces.
#       sum(.)     
#     ~^
```

i.e., the default setting has `EQ_SUB` (the `=` in a function call) increase the indentation level, and `%.>%` does so again.

`air` (0.11.0) reformats your code in a way that `indentation_linter()`'s defaults agree with:

```
list(
  "segment" = c(
    1,
    2
  ) %.>%
    sum(.)
)
```

So I think this is safe to close. Please re-open a new issue otherwise.
